Utilities/COCommitDescriptor.h

```diff
@@ -309,3 +309,36 @@ extern NSString *kCOCommitMetadataShortDescription;
  * See also COCommitDescription Localization documentation section.
  */
 extern NSString *kCOCommitMetadataShortDescriptionArguments;
+/**
+ * The optional key that indicates the command group UUID used to synthesize
+ * the commit.
+ *
+ * The value is a UUID string.
+ *
+ * When this key is set, two other keys are set:
+ *
+ * <list>
+ * <item>kCOCommitMetadataNodeOperationIdentifier</item>
+ * <item>kCOCommitMetadataNodeInversed</item>
+ * </list>
+ */
+extern NSString *kCOCommitMetadataNodeUUID;
+/**
+ * The optional key used to describe the commit operation (e.g. selective
+ * undo/redo or implicit undo due to a pattern undo track manipulation).
+ *
+ * The value is a string representing a commit descriptor identifier.
+ *
+ * When this key is set, -[COCommitDescriptor localizedShortDescription] will 
+ * synthesize a special description by interpolating both 
+ * kCOCommitMetadataShortDescription + kCOCommitMetadataShortDescriptionArguments 
+ * into the short description template for the operation identifier.
+ */
+extern NSString *kCOCommitMetadataNodeOperationIdentifier;
+/**
+ * The optional key that indicates whether the commit was created by inversing 
+ * the command referenced by kCOCommitMetadataNodeUUID.
+ *
+ * The value is a NSNumber boolean.
+ */
+extern NSString *kCOCommitMetadataNodeInversed;
```

Comment:
IMO the "Node" doesn't add much information, how about kCOCommitMetadataInversed ?